base/MAPL_Generic.F90

```diff
@@ -2128,8 +2141,13 @@ recursive subroutine MAPL_GenericFinalize ( GC, IMPORT, EXPORT, CLOCK, RC )
                                LABEL="INTERNAL_HEADER:", &
                                RC=STATUS)
         _VERIFY(STATUS)
-        call MAPL_ESMFStateWriteToFile(STATE%INTERNAL,CLOCK,FILENAME, &
-             FILETYPE, STATE, hdr/=0, oClients = o_Clients, RC=STATUS)
+        IF (FILENAME(1:1) == '-' .or. FILENAME(1:1) == '+') THEN
+           call MAPL_ESMFStateWriteToFile(STATE%INTERNAL,CLOCK,FILENAME(2:), &
+                FILETYPE, STATE, hdr/=0, oClients = o_Clients, RC=STATUS)
+        else
+           call MAPL_ESMFStateWriteToFile(STATE%INTERNAL,CLOCK,FILENAME, &
+                FILETYPE, STATE, hdr/=0, oClients = o_Clients, RC=STATUS)
+        endif
```

Comment:
These and the subsequent changes are (I don't think) adjoint specific. I was having a lot of weird issues with files with + signs at the begining which was supposed to signify something to MAPL but it was trying to open them including the + and failing. I'm not sure if we've stopped using the + in files, but if so, I can remove these changes.